src/test/java/org/assertj/core/extractor/ByNameSingleExtractorTest.java

```diff
@@ -176,4 +184,20 @@ public boolean isAdult() {
 	return new ByNameSingleExtractor<>("name");
   }
 
+  /** This style of Optional handling is emitted by Immutables code gen library. */
+  static class BareOptionalIntHolder {
+      private final Integer value;
+
+      BareOptionalIntHolder() {
+          value = null;
+      }
+
+      BareOptionalIntHolder(int value) {
+          this.value = value;
+      }
+
+      public OptionalInt value() {
+          return OptionalInt.of(value);
+      }
```

Comment:
what happens if you add  `public void value(int whatever)` ? I think the code to find the accessor method does not verify the method returns something and does not take any parameters.

Reply:
I'll extend the test to verify this case.  I don't think it's a problem - `Introspection.findMethod` uses `getMethod` / `getDeclaredMethod`.  It's not obvious from a quick look, but both these are varargs invocations with a zero-length parameter list.  So it will look for nullary methods only.